core/domain/topic_services.py

```diff
@@ -1389,6 +1389,28 @@ def check_can_edit_topic(
     return False
 
 
+def check_can_edit_question(
+    user: user_domain.UserActionsInfo,
+    topic_rights: Optional[topic_domain.TopicRights]
+) -> bool:
+    """Checks whether the user can edit the questions.
+
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        topic_rights: TopicRights or None. Rights object for the given topic.
+
+    Returns:
+        bool. Whether the given user can edit the question.
+    """
+    return topic_rights is not None and (
+        role_services.ACTION_EDIT_ANY_QUESTION in user.actions or (
+            role_services.ACTION_EDIT_QUESTION_IN_MANAGED_TOPIC in user.actions
+            and check_can_edit_topic(user, topic_rights)
```

Comment:
For readability, can we split this into separate if statements?

e.g.
```suggestion
    if topic_rights is None:
        return False
    if role_services.ACTION_EDIT_ANY_QUESTION in user.actions:
        return True
    if role_services.ACTION_EDIT_QUESTION_IN_MANAGED_TOPIC in user.actions:
        return check_can_edit_topic(user, topic_rights)
    return False
```

Reply:
> For readability, can we split this into separate if statements?
> 
> e.g.

I tried making these changes but it is affecting type checks and lint checks

Reply:
@KartikSuryavanshi I think you'll need to give more detail when responding to reviewers. If you run into an error, please include the full stack trace and explain what you've done to debug it. Thanks.

Reply:
Hey @seanlip, I’m sorry about that. I'll make sure to provide more details on this , including the full stack trace and the steps I've taken to debug the error.